### ISSUE TITLE
GPU support

### ISSUE THREAD

**Issue by adayton1** [sandialabs/verdict#4]
Are there any plans for adding GPU support to verdict? My GPU accelerated code is hitting a big slowdown when I have to switch to host only execution to call some verdict functions. It looks to me as if porting verdict would mostly involve adding `__host__ __device__` specifiers to functions and switching from std:: math functions to the corresponding c versions.

**Comment by clintonstimpson** [comment]
Hi @adayton1, we do not currently have plans to add GPU support to Verdict.  Though there is a chance we could soon use Verdict on the GPU ourselves.  Do you have specific changes you want to see?

**Comment by adayton1** [comment]
I use the hex and quad functions. But it's trivial enough to port the whole library. I put up pull request #5

**Comment by dtaller** [comment]
Hi. Do you know if the verdict pull request with GPU support, https://github.com/sandialabs/verdict/pull/5 will get merged in?

**Comment by clintonstimpson** [comment]
Yes, it will be merged, and was merged last week internally.  We'll be pushing a new update to the verdict source to this github repo soon.

**Comment by dtaller** [comment]
Thanks!

**Comment by liu15** [comment]
I've been testing Alan's implementation with ROCM on mi300a (rzadams) and found that the stack memory was getting exhausted for the hex_distortion (and quad_distortion) calls unless I specified 128K(!) gpu stack memory.  This was due to the arrays that are sized maxTotalNumberGaussPoints*maxNumberNodes and maxNumberNodes*maxNumberNodes.

For our use, we only have linear elements, so setting maxTotalNumberGaussPoints and maxNumberNodes to 8 fixed the issue for us, but obviously that is not a general solution.

I'm not sure what is the preferred "correct" fix: compile-time switch, multiple implementations, or a templated API.

**Comment by clintonstimpson** [comment]
GPU support is now in Verdict.  And we've also verified it works on rzadams.

@liu15, thanks for raising the stack memory issue.  I wasn't aware that was an issue until now.
I'm not yet sure what the best solution for that is either.
